.storybook/decorators/PropCombinations/README.md

```diff
@@ -0,0 +1,275 @@
+
+As an alternative, this custom decorator reads metadata from the storybook default export, and generates a single story that renders all combinations of props defined in the story metadata.
+
+# Getting started
+
+## Storybook Config
+
+Add a `generate` parameter in the story meta object.
+
+```tsx
+// Button.stories.tsx
+const meta: StoryMeta<typeof Button> = {
+  component: Button,
+  parameters: {
+    default: 'LiveExample',
+    // Define combinations of props to generate
+    generate: {
+      props: {
+        darkMode: [false, true],
+        rightGlyph: [undefined, <Icon glyph={'ArrowRight'} />],
+        leftGlyph: [undefined, <Icon glyph={'Cloud'} />],
+        children: ['MongoDB', undefined],
+        size: Object.values(Size),
+        variant: Object.values(Variant),
+      },
+    },
+  },
+};
+export default meta;
+```
+
+Export an empty story titled `Generated` from the story file
+
+```tsx
+// Button.stories.tsx
+...
+export const Generated = () => {};
+```
+
+This will create a story called "Generated", that renders all defined prop combinations.
+
+![Decorator Demo](./docs/decorator-demo.png)
+
+## Multiple generated stories
+
+You can also break up the generated prop combinations into multiple stories.
+
+To do this, list the generated story names in the `generate` parameter, and set the args you want to remain static on the individual story.
+
+```tsx
+const meta: StoryMeta<typeof Button> = {
+  ...
+  parameters: {
+    generate: {
+      stories: ['LargeSize', 'DefaultSize'],
+      props: { ... },
+    },
+  }
+}
+
+...
+
+export const DefaultSize: StoryType<typeof Button> = () => <></>;
+DefaultSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Default,
+    },
+  },
+};
+
+export const LargeSize: StoryType<typeof Button> = () => <></>;
+LargeSize.parameters = {
+  generate: {
+    args: {
+      size: Size.Large,
+    },
+  },
+};
+```
+
+# Options
+
+## `args`
+
+`Record<key in keyof ComponentProps, ComponentProps[key]>`
+
+By default the decorator pulls args from the main `meta.args` value. However, you may need to set args that are different for the generated instances. In this case,
+use `generate.args` to pass static props that are different from the main story args.
+
+```tsx
+const meta: StoryMeta<typeof Tooltip> = {
+  component: Tooltip,
+  parameters: {
+    default: 'LiveExample',
+    generate: {
+      props: { ... },
+      args: {open: true}
+    },
+  },
+};
+export default meta;
+```
+
+## `decorator`
+
+`(InstanceFn: StoryFn, context?: {args: Args}) => JSX.Element`
+
+The GeneratedStory decorator will _not_ read the decorators value defined in `meta.decorators`. To add a wrapper around each instance, define a decorator within `generate`.
```

Comment:
Why not?

Reply:
The `meta.decorators` object doesn't exist on the context we get in the decorator function. (My guess is because there would be a self reference in the object)?